### ISSUE TITLE
GitHub Actions CI Hanging Indefinitely

### ISSUE THREAD

**Issue by eseay** [airbnb/lottie-ios#2051]
## Which Version of Lottie are you using?

Lottie 4.1.3

## Expected Behavior 
App builds in GitHub Actions CI.

## Actual Behavior
We're seeing an issue where, after including Lottie as a dependency, our CI builds hang indefinitely at the end of package resolution. 

Based on some research, we've determined that this issue is related to the [inclusion of a Swift Package plugin](https://github.com/airbnb/lottie-ios/blob/master/Package.swift#L10-L13) in Lottie's package manifest. We believe that at build time, the system is prompting for user input to allow the plugin to run, but when that happens, obviously there is no one to do that input since it's a build machine.

We've attempted to disable package plugin validation by setting some defaults as well as passing an additional argument to `xcodebuild`, but we're not seeing that have any effect.

```
defaults write com.apple.dt.Xcode IDESkipPackagePluginFingerprintValidatation -bool YES
set -o pipefail && xcodebuild test -scheme THE_SCHEME -destination 'name=iPhone 14 Pro' -skipPackagePluginValidation -verbose
```

We're seeing the same issue when invoking tests from Fastlane too (though, that obviously is just sugar around the same fundamentals).

I have to imagine that others have seen and worked around this issue before, but I can't seem to find any definitive solutions online.

**Comment by eseay** [comment]
@calda I will work on trying to replicate this in a sandbox project. Stay tuned. (I am using lottie-spm already, which makes this even stranger.)

**Comment by calda** [comment]
If you're using lottie-spm then this can't be related to the Swift Package Plugin in lottie-ios's `Package.swift` -- lottie-spm's `Package.swift` doesn't include the Swift Package Plugin

**Comment by calda** [comment]
Something else to check then -- does the issue go away if you switch from lottie-spm _back_ to lottie-ios?

**Comment by eseay** [comment]
> Something else to check then -- does the issue go away if you switch from lottie-spm _back_ to lottie-ios?

Hmm 🤔 ... in fact, it *does go away* when swapping back. I honestly don't know what to make of that. I've swapped back and forth a few times and can confirm that every time I switch to lottie-spm, it doesn't work, and every time I go back to lottie-ios, it works.

The great news is that I seem to be unblocked; the bad news is that I still have no idea why Lottie SPM would have not been working, especially if you feel confident in ruling out package plugins as a culprit.

**Comment by calda** [comment]
It must be an issue with Xcode / Swift Package Manager. I'd suggest filing a feedback with Apple.

I'm going to close this for now since there probably wouldn't be a fix for this on our end. I'll keep a look out for anyone else experiencing similar issues though.

**Comment by eseay** [comment]
@calda Thanks - that works for me.

**Comment by eseay** [comment]
Hey @calda I wanted to follow back up on this and share that we did manage to find a solution in collaboration with GitHub support. The issue stemmed from the command line not being able to properly authenticate with GitHub.com and api.github.com when trying to get binaries.

I have added the following step to my workflows (note that the actual secret key will differ), and everything is now working as expected! Perhaps something about this could be added to the docs in case anyone else runs into this in the future.

```
- name: Add .netrc
  run: |
    echo -e "machine github.com\n  login oauth2\n  password ${{ secrets.GITHUB_TOKEN }}\n" > ~/.netrc
    echo -e "machine api.github.com\n  login oauth2\n  password ${{ secrets.GITHUB_TOKEN }}\n" >> ~/.netrc
  shell: sh
```
